cpp/src/transform/encode.cu

```diff
@@ -0,0 +1,83 @@
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/detail/gather.hpp>
+#include <cudf/detail/null_mask.hpp>
+#include <cudf/detail/search.hpp>
+#include <cudf/detail/sorting.hpp>
+#include <cudf/detail/stream_compaction.hpp>
+#include <cudf/detail/transform.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/utilities/bit.hpp>
+#include <numeric>
+
+namespace cudf {
+namespace detail {
+
+std::pair<std::unique_ptr<table>, std::unique_ptr<column>> encode(
+  table_view const& input_table, rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+{
+  std::vector<size_type> columns(input_table.num_columns());
```

Comment:
`columns` is a little ambiguous here. Can we name it something more specific? At first glance I thought it was a vector of columns. :)

Reply:
Changed to `drop_keys`